### PR TITLE
tests: Fix various test failures in 32-bit environment

### PR DESCRIPTION
**Pull request by honggyukim**
This PR mainly fixes the problems when testing i386 under x86_64.

It fixes tests in i386 under x86_64 environment:
```
043 full_demangle       : NG NG NG NG NG NG NG NG NG NG
058 arg_int             : NG NG NG NG NG SK SK SK SK SK
063 retval              : NG NG NG NG NG SK SK SK SK SK
065 arg_order           : NG NG NG NG NG SK SK SK SK SK
083 arg_float           : NG NG NG NG NG SK SK SK SK SK
084 arg_mixed           : NG NG NG NG NG SK SK SK SK SK
086 arg_stack           : NG NG NG NG NG SK SK SK SK SK
087 arg_variadic        : NG NG NG NG NG SK SK SK SK SK
```
It fixes tests in arm 32-bit environment:
```
048 malloc_impl         : TM TM TM TM TM TM TM TM TM TM
058 arg_int             : NG NG NG NG NG SK SK SK SK SK
063 retval              : NG NG NG NG NG SK SK SK SK SK
065 arg_order           : NG NG NG NG NG SK SK SK SK SK
119 malloc_hook         : TM TM TM TM TM TM TM TM TM TM
120 malloc_tsd          : TM TM TM TM TM TM TM TM TM TM
121 malloc_fork         : TM TM TM TM TM TM TM TM TM TM
```
Fixed: #1083

Signed-off-by: Honggyu Kim <honggyu.kp@gmail.com>

### CODE REVIEW

**Comment by honggyukim** [comment]
This PR fixes all the tests reported in #1083 except for `085 arg_reg`.

**Comment by honggyukim** [comment]
We may have to apply the following change for `085 arg_reg`, but not sure because it's actually a test for stack not for register.
```diff
diff --git a/tests/t085_arg_reg.py b/tests/t085_arg_reg.py
index 223d0f3..cc072ae 100644
--- a/tests/t085_arg_reg.py
+++ b/tests/t085_arg_reg.py
@@ -36,5 +36,11 @@ class TestCase(TestBase):
             argopt = argopt.replace('32%xmm0', '32%s0')
             argopt = argopt.replace('%xmm0', '%d0')
             argopt = argopt.replace('fparg1/80%stack+1', 'fparg1/80')
+        elif TestBase.is_32_bit_elf(self):
+            argopt  = '-A "mixed_add@arg1/i32%stack+1,fparg1/32%stack+2" '
+            argopt += '-A "mixed_sub@arg1/x%stack+1,arg2%stack+2" '
+            argopt += '-A "mixed_mul@fparg1%stack+1,arg2/i64%stack+3" '
+            argopt += '-A "mixed_div@arg1/i64,fparg1/80%stack+3" '
+            argopt += '-A "mixed_str@arg1/s%stack+1,fparg1%stack+2"'

         return '%s %s %s' % (TestBase.uftrace_cmd, argopt, 't-' + self.name)
```
It makes the test `085 arg_reg` pass anyway.

File: tests/t065_arg_order.py [hunk 1]
```diff
@@ -26,8 +26,8 @@ def runcmd(self):
 
         if TestBase.is_32_bit_elf(self):
             # int_mul@arg1 is a 'long long', so we should skip arg2
-            argopt  = '-A "int_mul@arg1/i64" -A "^int_@arg1" '
-            argopt += '-A "int_(add|sub|div)@arg2" -A "int_mul@arg3/x" '
+            argopt  = '-A "int_mul@arg3/x" -A "^int_@arg1" '
+            argopt += '-A "int_(add|sub|div)@arg2" -A "int_mul@arg1/i64" '
             argopt += '-A "int_add@arg1/i32"'
```
Comment (namhyung) [inline tests/t065_arg_order.py]: Then I think the last line `-A int_add@arg1/i32` is not necessary.

Comment (honggyukim) [reply tests/t065_arg_order.py]: Thanks. I just moved the line.

**Comment by honggyukim** [comment]
@namhyung If you prefer to change the name `is_32_bit_elf` to `is_elfclass32`, then I will change it as well.

Likewise, I'm thinking of adding another helper function `get_elf_machine`, which returns `e_machine` type in the given ELF file. It'd be better than checking the architecture as follows:
```
$ grep "startswith" * | grep arm
t051_return.py:        if platform.machine().startswith('arm'):
t084_arg_mixed.py:        if platform.machine().startswith('arm'):
t085_arg_reg.py:        if platform.machine().startswith('arm'):
t136_dynamic.py:        if platform.machine().startswith('arm'):
t140_dynamic_xray.py:        if platform.machine().startswith('arm'):
t223_dynamic_full.py:        if platform.machine().startswith('arm'):
t232_dynamic_unpatch.py:        if platform.machine().startswith('arm'):
t233_dynamic_unpatch2.py:        if platform.machine().startswith('arm'):
```
```
$ grep "platform.architecture" *
t043_full_demangle.py:        if platform.architecture()[0].startswith('32bit'):
t058_arg_int.py:        if platform.architecture()[0].startswith('32bit'):
t063_retval.py:        if platform.architecture()[0].startswith('32bit'):
t065_arg_order.py:        if platform.architecture()[0].startswith('32bit'):
t097_dump_basic.py:        if platform.architecture()[0] == '32bit':
t098_dump_tid.py:        if platform.architecture()[0] == '32bit':
t099_dump_filter.py:        if platform.architecture()[0] == '32bit':
t100_dump_depth.py:        if platform.architecture()[0] == '32bit':
```

**Comment by namhyung** [comment]
I think `is_32bit` is just fine.

**Comment by honggyukim** [comment]
I changed `is_32_bit_elf` to `is_32bit`.

**Comment by honggyukim** [comment]
@namhyung What do you think about adding the change in https://github.com/namhyung/uftrace/pull/1089#issuecomment-574953526?

**Comment by namhyung** [comment]
It'd be better skipping it when i386 doesn't pass it in registers.

File: tests/runtest.py [hunk 2]
```diff
@@ -371,6 +371,40 @@ def check_perf_paranoid(self):
...
+        # e_ident[] indexes
+        EI_MAG0       = 0
+        EI_MAG1       = 1
+        EI_MAG2       = 2
+        EI_MAG3       = 3
+        EI_CLASS      = 4
+        EI_DATA       = 5
+        EI_VERSION    = 6
+        EI_OSABI      = 7
+        EI_PAD        = 8
```
Comment (namhyung) [inline tests/runtest.py]: We don't need all of these, just keep `EI_CLASS` here.

Comment (honggyukim) [reply tests/runtest.py]: I removed them.

**Comment by honggyukim** [comment]
I've made `get_elf_machine()` to get its e_machine in string format.

> It'd be better skipping it when i386 doesn't pass it in registers.

I've made t085 test skiped only in i386 machine with `get_elf_machine()`.

@namhyung If you agree, I will change all the other arch check routines with `platform` to this way. Please see https://github.com/namhyung/uftrace/pull/1089#issuecomment-576352737.

**Comment by namhyung** [comment]
I'm ok with the machine change too. :)

**Comment by honggyukim** [comment]
I applied the change as follows:

ChangeLog:
-  add 'tests: Fix malloc tests in ARMv7(raspberry pi)' in https://github.com/namhyung/uftrace/pull/1089/commits/8ed096c1747b6d16f68561d804819a04f06a2b6c
- add 'tests: Replace platform.machine to TestBase.get_elf_machine' in https://github.com/namhyung/uftrace/pull/1089/commits/540e4e11d59d5bf6ecf3507fb9cc4103fd6450f2

**Comment by honggyukim** [comment]
Here is "failed-tests.txt" result in i386 mode.
```
Test case                 pg             finstrument-fu
------------------------: O0 O1 O2 O3 Os O0 O1 O2 O3 Os
012 demangle            : NG NG NG NG NG NG NG NG NG NG
052 nested_func         : OK OK OK OK OK NG NG NG NG NG
122 time_range2         : NG NG NG NG NG OK OK OK OK OK
124 exception           : NG NG NG NG NG NG NG NG NG NG
136 dynamic             : BI BI BI BI BI BI BI BI BI BI
140 dynamic_xray        : BI BI BI BI BI BI BI BI BI BI
147 event_sdt           : BI BI BI BI BI BI BI BI BI BI
150 recv_event          : BI BI BI BI BI BI BI BI BI BI
151 recv_runcmd         : NG NG NG NG NG NG NG NG NG NG
161 pltbind_now         : NG NG NG NG NG NG NG NG NG NG
162 pltbind_now_pie     : NG NG NG NG NG NG NG NG NG NG
182 thread_exit         : OK OK OK OK NG OK OK OK OK OK
184 arg_enum            : NG NG NG NG NG NG NG NG NG NG
185 exception2          : NG NG NG NG NG NG NG NG NG NG
186 exception3          : NG NG NG NG NG NG NG NG NG NG
201 arg_dwarf1          : NG OK OK OK OK SK SK SK SK SK
202 arg_dwarf2          : NG SG SG SG SG SK SK SK SK SK
203 arg_dwarf3          : OK OK NG NG NG SK SK SK SK SK
204 arg_dwarf4          : NG NG NG NG NG SK SK SK SK SK
205 arg_auto4           : NG NG NG NG NG SK SK SK SK SK
212 noplt_libcall       : BI BI BI BI BI BI BI BI BI BI
213 arg_symbol          : NG NG NG NG NG SK SK SK SK SK
219 no_libcall_script   : NZ NZ NZ NZ NZ NZ NZ NZ NZ NZ
220 trace_script        : NG NG NG NG NG NG NG NG NG NG
223 dynamic_full        : BI BI BI BI BI BI BI BI BI BI
224 dynamic_lib         : SG SG SG SG SG SG SG SG SG SG
225 dynamic_size        : SG SG SG SG SG SG SG SG SG SG
228 read_pmu_cycle3     : OK OK NG NG NG OK OK OK OK OK
232 dynamic_unpatch     : BI BI BI BI BI BI BI BI BI BI
233 dynamic_unpatch2    : BI BI BI BI BI BI BI BI BI BI
```
Here is "failed-tests.txt" result in ARM 32-bit machine.
```
Test case                 pg             finstrument-fu
------------------------: O0 O1 O2 O3 Os O0 O1 O2 O3 Os
012 demangle            : NG NG NG NG NG NG NG NG NG NG
043 full_demangle       : NG NG NG NG NG NG NG NG NG NG
052 nested_func         : OK OK OK OK OK NG NG NG NG NG
082 arg_many            : NG NG NG NG NG SK SK SK SK SK
083 arg_float           : NG NG NG NG NG SK SK SK SK SK
084 arg_mixed           : NG NG NG NG NG SK SK SK SK SK
085 arg_reg             : NG NG NG NG NG SK SK SK SK SK
086 arg_stack           : NG OK OK NG OK SK SK SK SK SK
087 arg_variadic        : NG NG NG NG NG SK SK SK SK SK
124 exception           : NG NG NG NG NG NG NG NG NG NG
131 lib_dlopen          : NZ NZ NZ NZ NZ OK OK OK OK OK
136 dynamic             : BI BI BI BI BI BI BI BI BI BI
140 dynamic_xray        : BI BI BI BI BI BI BI BI BI BI
147 event_sdt           : BI BI BI BI BI BI BI BI BI BI
150 recv_event          : BI BI BI BI BI BI BI BI BI BI
151 recv_runcmd         : NG NG NG NG NG NG NG NG NG NG
184 arg_enum            : NG NG NG NG NG NG NG NG NG NG
185 exception2          : NG NG NG NG NG NG NG NG NG NG
186 exception3          : SG SG SG SG SG NG NG NG NG NG
198 lib_arg_float       : NG NG NG NG NG SK SK SK SK SK
200 lib_dlopen2         : NZ NZ NZ NZ NZ OK OK OK OK OK
201 arg_dwarf1          : NG NG NG NG NG SK SK SK SK SK
202 arg_dwarf2          : NG NG NG NG NG SK SK SK SK SK
203 arg_dwarf3          : NG NG NG NG NG SK SK SK SK SK
224 dynamic_lib         : NZ NZ NZ NZ NZ NZ NZ NZ NZ NZ
225 dynamic_size        : NZ NZ NZ NZ NZ NZ NZ NZ NZ NZ
232 dynamic_unpatch     : BI BI BI BI BI BI BI BI BI BI
```

**Pull request merged**